src/utils/arrow.rs

```diff
@@ -71,3 +74,42 @@ pub(crate) fn arrow_schema_from_column_names_and_range(
 pub(crate) fn record_batch_to_pybytes<'p>(py: Python<'p>, rb: &RecordBatch) -> Result<&'p PyBytes> {
     record_batch_to_bytes(rb).map(|bytes| PyBytes::new(py, bytes.as_slice()))
 }
+
+/// Arrow array to Python.
+pub(crate) fn to_python_array(
+    array: &ArrayRef,
+    py: Python,
+    pyarrow: &PyModule,
+) -> PyResult<PyObject> {
+    let ffi_array = ArrowArray::try_new(array.data().to_owned()).with_context(|| "haha")?;
+    let (array_ptr, schema_ptr) = ArrowArray::into_raw(ffi_array);
+    let array = pyarrow.getattr("Array")?.call_method1(
+        "_import_from_c",
+        (array_ptr as Py_uintptr_t, schema_ptr as Py_uintptr_t),
+    )?;
```

Comment:
Hmmm is it really okay to use an internal arrow interface for this ? We might end up putting very restricting constraints on the `pyarrow` version, which might be annoying for end users. Except if it's a "public private" interface à la `NamedTuple._replace`

Reply:
it seems from what i read [there](https://arrow.apache.org/docs/format/CDataInterface.html) to be very consistent.
> Allow zero-copy sharing of Arrow data between independent runtimes and components running in the same process.

Maybe i miss-understood ?

Reply:
Just double-checked and the functioin is mentionned in the docs so we can consider it won't move :+1: https://arrow.apache.org/docs/python/generated/pyarrow.RecordBatchReader.html .

Could you please add a comment pointing to that and explaining that this functiuon is not really private ?